pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -856,14 +856,40 @@ func (cloud *CloudProvider) WaitForAttach(ctx context.Context, project string, v
 }
 
 func wrapOpErr(name string, opErr *computev1.OperationErrorErrors) error {
```

Comment:
Also what about error handling for cases where the operation is not even initiated, and we encounter error in say insertDisk, attachDisk where the WaitForOp is not called. e.g [code](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/pkg/gce-cloud-provider/compute/gce-compute.go#L724)

Also we may need to handle uncastable error (i.e errors.As cannot case to gce.UnsupportedDiskError) like the ones we saw in filestore. May need some manual testing here (1. insert disk when quota is exhausted, 2. attachdisk when disk already attached) or we can address them lazily if we encounter them.

Reply:
I have looked into quiet a few clusters, and I haven't seen any errors that are incorrectly returned as an Internal error at this point ( where the operation is not even initiated). I will look into some more clusters, but I have probably looked into at least 20 at this point, and I haven't seen errors like that. 

> Also we may need to handle uncastable error (i.e errors.As cannot cast to gce.UnsupportedDiskError)  like the ones we saw in filestore. 

I'm a little confused by this comment. Are you saying we could be missing some cases where we need to  [return an UnsupportedDiskError](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/26fe3898b13083a02e28fa5dd16410c15136e2c5/pkg/gce-cloud-provider/compute/gce-compute.go#L859) (which we do when the operation error has the UNSUPPORTED_OPERATION code). This won't happen like in filestore because wrapOpErr passes in an operation error of type computev1.OperationErrorErrors directly. This error type doesn't actually implement the error interface, so this will never happen. (errors.As cannot even be run on the operation error of type computev1.OperationErrorErrors since it doesn't implement error interface).

Reply:
Ack. I missed the OperationErrorErrors type. 

And yes, for unsupported operations, I think we would return kIternal for the UnsupportedDiskError. Until we repro this scenario, it would be difficult to handle this error. So that would be a followup for future to understand the sceario better.

As discussed, lets capture the known issues in some task:
1. AttachDisk errors not handled by errors.As() [code](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/release-1.10/pkg/gce-pd-csi-driver/controller.go#L638)
2. [InsertDisk](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/release-1.10/pkg/gce-pd-csi-driver/controller.go#L1781), [DeleteDisk](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/release-1.10/pkg/gce-pd-csi-driver/controller.go#L480), [DetachDisk](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/release-1.10/pkg/gce-pd-csi-driver/controller.go#L758) errors are not handled.